src/microbe_stage/Microbe.Contact.cs

```diff
@@ -1304,6 +1304,13 @@ private void CheckStartEngulfingOnCandidates()
         // foreach (var microbe in touchedMicrobes.Concat(otherMicrobesInEngulfRange))
         foreach (var microbe in touchedMicrobes)
         {
+            if (microbe.destroyed)
+            {
+                GD.Print($"Removed destroyed microbe from {nameof(touchedMicrobes)}");
+                touchedMicrobes.Remove(microbe);
+                break; // this is important to not crash on the next loop in the foreach
```

Comment:
I didn't mean to literally include my explanation comment like this. If you want to keep this please put it on a separate line.